Week1/homework/js-exercises/JS2_w1e5/JS2_w1e5_theCatWalk.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+let cat = document.getElementsByTagName('img')[0]; //2
+cat.style.left = '0px'; //3
+let moveFwd = true;
+let leftValue = 0;
+
+function catWalk() { //4-
+  leftValue = parseInt(cat.style.left);
+  if (moveFwd && (leftValue > (window.innerWidth - cat.width))) moveFwd = false; //6-
+  if (!moveFwd && (leftValue <= 0)) moveFwd = true; //-6-
+  if (moveFwd) cat.style.left = (leftValue += 10) + 'px'; //-4
+  else cat.style.left = (leftValue -= 10) + 'px'; //-6
+
+  if ((leftValue >= (window.innerWidth / 2)) && (leftValue <= window.innerWidth)) cat.src = 'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif?itemid=10561424';//7*
```

Comment:
Very nice! Only a small notice, for having the change of the gif on the centre of the screen.
The `if` statement should be as follows:
`if (leftValue >= (window.innerWidth / 2) - (cat.width / 2))`

Also, as you see, the `leftValue <= window.innerWidth` doesn't contribute to the statement. You have already made this check above.